wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/Reactions/ReactionToggle.swift

```diff
@@ -0,0 +1,119 @@
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+import WireCommonComponents
+
+// MARK: - ReactionToggle
+
+class ReactionToggle: UIControl {
+
+    // MARK: - Properties
+
+    typealias ButtonColors = SemanticColors.Button
+
+    private let emojiLabel = DynamicFontLabel(fontSpec: .mediumRegularFont,
+                                              color: SemanticColors.Label.textDefault)
+    private let counterLabel = DynamicFontLabel(fontSpec: .mediumSemiboldFont,
+                                                color: SemanticColors.Label.textDefault)
+
+    private var onToggle: (() -> Void)?
+
+    var isToggled: Bool {
+        didSet {
+            guard oldValue != isToggled else { return }
+            updateAppearance()
+        }
+    }
+
+    // MARK: - Lifecycle
+
+    init(isToggled: Bool = false) {
+        self.isToggled = isToggled
+        super.init(frame: .zero)
+
+        layer.borderWidth = 1
+        layer.cornerRadius = 12
+        layer.masksToBounds = true
+
+        let stackView = UIStackView(arrangedSubviews: [emojiLabel, counterLabel])
+        stackView.axis = .horizontal
+        stackView.distribution = .fillEqually
+        stackView.alignment = .center
+        stackView.spacing = 4
+
+        stackView.isLayoutMarginsRelativeArrangement = true
+        stackView.layoutMargins = UIEdgeInsets(top: 4, left: 8, bottom: 4, right: 0)
+
+        stackView.isUserInteractionEnabled = false
+
+        addSubview(stackView)
+
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+
+        NSLayoutConstraint.activate([
+            stackView.topAnchor.constraint(equalTo: topAnchor),
+            stackView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            stackView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            stackView.bottomAnchor.constraint(equalTo: bottomAnchor)
+        ])
+
+        updateAppearance()
+        addTarget(self, action: #selector(didToggle), for: .touchUpInside)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Methods
+
+    public func configureData(
+        type: String,
+        count: Int,
+        isToggled: Bool,
+        onToggle: @escaping () -> Void
+    ) {
+        emojiLabel.text  = type
+        counterLabel.text = String(count)
+        self.isToggled = isToggled
+        self.onToggle = onToggle
+    }
+
+    private func updateAppearance() {
+        backgroundColor = isToggled
+        ? ButtonColors.backgroundReactionSelected :
+        ButtonColors.backroundReactionNormal
```

Comment:
Don't go to next line in middle of ternary
```suggestion
        backgroundColor = isToggled ? ButtonColors.backgroundReactionSelected : ButtonColors.backroundReactionNormal
```